### PR TITLE
chore: update `yeoman-generator` to v5

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
chore

**Did you add tests for your changes?**
No.

**If relevant, did you update the documentation?**
No.

**Summary**
update `yeoman-generator` to v5.

**Does this PR introduce a breaking change?**
No.

**Other information**
No/

### CODE REVIEW

File: packages/generators/src/addon-generator.ts [hunk 1]
```diff
@@ -6,6 +6,9 @@ import { CustomGenerator } from "./types";
 import type { CustomGeneratorOptions, BaseCustomGeneratorOptions } from "./types";
 import { getInstaller, getTemplate } from "./utils/helpers";
 
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+Object.assign(Generator.prototype, require("yeoman-generator/lib/actions/install"));
+
```
Comment (snitin315) [inline packages/generators/src/addon-generator.ts:11]: >Install action is deprecated and is not included by default

[Breaking change](https://github.com/yeoman/generator/releases/tag/v5.0.0)

File: packages/generators/src/init-generator.ts [hunk 2]
```diff
@@ -19,20 +19,6 @@ export default class InitGenerator<
...
-          "ℹ INFO ",
-        )} supplied generation path doesn't exist, required folders will be created.`,
-      );
-      try {
-        mkdirSync(this.generationPath, { recursive: true });
-      } catch (error) {
-        this.cli.logger.error(`Failed to create directory.\n ${error}`);
-        process.exit(2);
-      }
-    }
```
Comment (snitin315) [inline packages/generators/src/init-generator.ts:34]: This is now handled by the yeoman-generator. I've updated the test cases accordingly.

File: packages/generators/src/addon-generator.ts [hunk 3]
```diff
@@ -6,6 +6,9 @@ import { CustomGenerator } from "./types";
 import type { CustomGeneratorOptions, BaseCustomGeneratorOptions } from "./types";
 import { getInstaller, getTemplate } from "./utils/helpers";
 
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+Object.assign(Generator.prototype, require("yeoman-generator/lib/actions/install"));
```
Comment (snitin315) [inline packages/generators/src/addon-generator.ts:10]: >Install action is deprecated and is not included by default

[Breaking change](https://github.com/yeoman/generator/releases/tag/v5.0.0)

**Review by alexander-akait — APPROVED** [review]
Looks good, please rebase

**Comment by snitin315** [comment]
Rebased.

**Pull request merged**